Controllers/ProductsController.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class ProductsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public ProductsController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Products
+        public async Task<IActionResult> Index()
+        {
+            var Products = await _context.Products
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .ToListAsync();
+
+            var branchesVM = new List<ProductFormViewModel>();
+
+            foreach (var product in Products)
+            {
+                branchesVM.Add(_mapper.Map<ProductFormViewModel>(product));
+            }
+
+            return View(branchesVM);
```

Comment:
Try to simplify this to:
```c#
return View(_mapper.Map<IEnumerable<ProductFormViewModel>>(Products);
```
I guess it should work.
Please if succeed with this mapping simplification, apply this in all places in repo as this pattern show up often.